storehaus-algebra/src/main/scala/com/twitter/storehaus/algebra/BufferingStore.scala

```diff
@@ -57,13 +59,14 @@ class BufferingMergeable[K, V](store: Mergeable[K, V], summerCons: SummerConstru
         k -> foptV
       }
 
-  override def multiMerge[K1 <: K](kvs: Map[K1, V]) = {
-    val links: Map[K, PromiseLink[V]] = kvs.map { case (k1, v) => k1 -> PromiseLink(v) }(breakOut) // no lazy
-    summer.put(links).foreach(mergeFlush(_))
+  override def multiMerge[K1 <: K](kvs: Map[K1, V]): Map[K1, Promise[Option[V]]] = {
```

Comment:
this should be `Map[K1, Future[Option[V]]` we should definitely not expose the `Promise` which is actually mutable.